testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m

```diff
@@ -62,23 +69,123 @@ - (void)testPointerButtons {
+        assertOneMessageAndGetSequenceNo(messages, @"PointerChange.add,device=2,buttons=0");
+    rightClickDownSeqNo =
+        assertOneMessageAndGetSequenceNo(messages, @"PointerChange.down,device=2,buttons=2");
+    rightClickUpSeqNo =
+        assertOneMessageAndGetSequenceNo(messages, @"PointerChange.up,device=2,buttons=2");
+  }
+  XCTAssertGreaterThan(rightClickDownSeqNo, rightClickAddedSeqNo,
+                       @"Right-click pointer was down before add");
+  XCTAssertGreaterThan(rightClickUpSeqNo, rightClickDownSeqNo,
+                       @"Right-click pointer was up before down");
+  XCTAssertGreaterThan([[hoverSeqNos firstObject] intValue], 3,
+                       @"Right-click pointer caused hover before hover pointer add");
+  XCTAssertGreaterThan(hoverRemovedSeqNo, [[hoverSeqNos lastObject] intValue],
+                       @"Right-click pointer's hover pointer had hover event after it was removed");
+}
+
+- (void)testPointerHover {
+  BOOL supportsPointerInteraction = NO;
+  SEL supportsPointerInteractionSelector = @selector(supportsPointerInteraction);
+  if ([XCUIDevice.sharedDevice respondsToSelector:supportsPointerInteractionSelector]) {
+    supportsPointerInteraction =
+        performBoolSelector(XCUIDevice.sharedDevice, supportsPointerInteractionSelector);
+  }
+  XCTSkipUnless(supportsPointerInteraction, "Device does not support pointer interaction.");
+  XCUIApplication* app = [[XCUIApplication alloc] init];
+  app.launchArguments = @[ @"--pointer-events" ];
+  [app launch];
+
+  NSPredicate* predicateToFindFlutterView =
+      [NSPredicate predicateWithBlock:^BOOL(id _Nullable evaluatedObject,
+                                            NSDictionary<NSString*, id>* _Nullable bindings) {
+        XCUIElement* element = evaluatedObject;
+        return [element.identifier hasPrefix:@"flutter_view"];
+      }];
+  XCUIElement* flutterView = [[app descendantsMatchingType:XCUIElementTypeAny]
+      elementMatchingPredicate:predicateToFindFlutterView];
+  if (![flutterView waitForExistenceWithTimeout:kSecondsToWaitForFlutterView]) {
+    NSLog(@"%@", app.debugDescription);
+    XCTFail(@"Failed due to not able to find any flutterView with %@ seconds",
+            @(kSecondsToWaitForFlutterView));
+  }
+
+  XCTAssertNotNil(flutterView);
+
+  SEL hover = @selector(hover);
+  XCTAssertTrue([flutterView respondsToSelector:hover],
+                @"If supportsPointerInteraction is true, this should be true too.");
+  [flutterView performSelector:hover];
+  XCTAssertTrue(
+      [app.textFields[@"0,PointerChange.add,device=0,buttons=0"] waitForExistenceWithTimeout:1],
+      @"PointerChange.add event did not occur for a hover");
+  XCTAssertTrue(
+      [app.textFields[@"1,PointerChange.hover,device=0,buttons=0"] waitForExistenceWithTimeout:1],
+      @"PointerChange.hover event did not occur for a hover");
+  // The number of hover events fired is not always the same
+  int i = 2;
+  while (
```

Comment:
I don't understand what the difference in behaviour would be, how would `NSPredicate` act?

Reply:
Sorry, I didn't get a chance to get back to this yesterday.
I was thinking of using a predicate that accepted `"%d,PointerChange.hover,device=0,buttons=0` with %d being any number, instead of checking every number in a `while` loop.
At the least, I recommend an upper limit for this `while` so the test eventually fails instead of eventually overflowing and hanging.